### PR TITLE
chore: add gatsbyjs example

### LINKED ISSUE #117: Add example: `with-gatsby`
## Description

Add the simplest possible example of how to use this library with [gatsby](https://github.com/gatsbyjs/gatsby).

This issue is open to anyone and could serve as a fun introduction to `remirror` 😄 

## Possible Implementation

Take a look at the current [`with-next`](https://github.com/ifiokjr/remirror/tree/master/examples/with-next) and [`with-razzle`](https://github.com/ifiokjr/remirror/tree/master/examples/with-razzle) implementations.

### PR DESCRIPTION
**Pull request by alexluong**
Closes #117

## Description

I added the Gatsby example. However, there were some ESlint problems. Can you take a look @ifiokjr?

Screen Shot 2019-08-08 at 10 27 26 PM

### CODE REVIEW

**Comment by ifiokjr** [comment]
@alexluong Looks good to me and the build also passes. 

I can't see the lint issues which you're referencing. It looks like it might be local eslint config picking up the `lib` and `dist` folders during the build.

I'm happy to merge it as it is.

File: examples/with-gatsby/package.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+  "private": true,
+  "name": "with-gatsby",
+  "license": "MIT",
+  "version": "0.4.2-canary.2",
+  "scripts": {
+    "start": "gatsby develop",
+    "develop": "gatsby develop",
+    "build": "gatsby build",
```
Comment (ifiokjr) [inline examples/with-gatsby/package.json]: Optional: add a lint rule here.

```suggestion
    "build": "gatsby build",
    "lint": "tslint --project tsconfig.lint.json --config ../../tslint.json",
```

**Comment by alexluong** [comment]
@ifiokjr Did you try pulling down this PR and run the Gatsby site locally? It's not working for me there.

**Comment by ifiokjr** [comment]
> @ifiokjr Did you try pulling down this PR and run the Gatsby site locally? It's not working for me there.

@alexluong I haven't actually. 

I'll check it out once I'm in front of my laptop.

**Comment by ifiokjr** [comment]
@alexluong just verified the issue you're facing. 

Adding the following file `examples/with-gatsby/.eslintignore` should fix it. 

```
lib
**/lib/**
dist
node_modules
public
```

Thanks for sorting this out 👍

**Comment by ifiokjr** [comment]
@alexluong I know it's been a while, but any chance you could build this again and test if it works for you. The project has switched over to `eslint` so I'm thinking it might just work.

**Pull request closed without merging**